tests/nvapi_vulkan.cpp

```diff
@@ -0,0 +1,243 @@
+
+            REQUIRE(NvAPI_Vulkan_InitLowLatencyDevice(vkDevice.get(), &signalSemaphoreHandle) == NVAPI_OK);
+            REQUIRE(NvAPI_Vulkan_Sleep(vkDevice.get(), signal) == NVAPI_OK);
+            REQUIRE(NvAPI_Vulkan_DestroyLowLatencyDevice(vkDevice.get()) == NVAPI_OK);
+        }
+
+        SECTION("GetLatency returns OK") {
+            REQUIRE_CALL(*vkDevice, vkGetLatencyTimingsNV(_, _, _))
+                .SIDE_EFFECT({
+                    _3->pTimings[0].presentID = 5;
+                    _3->pTimings[0].inputSampleTimeUs = 8;
+                    _3->pTimings[1].presentID = 6;
+                    _3->pTimings[1].inputSampleTimeUs = 7;
+                })
+                .RETURN(VK_SUCCESS);
+
+            SetupResourceFactory(std::move(dxgiFactory), std::move(vk), std::move(nvml), std::move(lfx));
+
+            REQUIRE(NvAPI_Vulkan_InitLowLatencyDevice(vkDevice.get(), &signalSemaphoreHandle) == NVAPI_OK);
+
+            NV_VULKAN_LATENCY_RESULT_PARAMS params{};
+            params.version = NV_VULKAN_LATENCY_RESULT_PARAMS_VER1;
+            REQUIRE(NvAPI_Vulkan_GetLatency(vkDevice.get(), &params) == NVAPI_OK);
+            REQUIRE(params.frameReport[0].frameID == 5);
+            REQUIRE(params.frameReport[0].inputSampleTime == 8);
+            REQUIRE(params.frameReport[1].frameID == 6);
+            REQUIRE(params.frameReport[1].inputSampleTime == 7);
+
+            REQUIRE(NvAPI_Vulkan_DestroyLowLatencyDevice(vkDevice.get()) == NVAPI_OK);
+        }
+
+        SECTION("GetLatency with unknown struct version returns incompatible-struct-version") {
+            NV_VULKAN_LATENCY_RESULT_PARAMS params;
+            params.version = NV_VULKAN_LATENCY_RESULT_PARAMS_VER1 + 1;
+            REQUIRE(NvAPI_Vulkan_GetLatency(vkDevice.get(), &params) == NVAPI_INCOMPATIBLE_STRUCT_VERSION);
+        }
+
+        SECTION("GetLatency with current struct version returns not incompatible-struct-version") {
+            // This test should fail when a header update provides a newer not yet implemented struct version
+            NV_VULKAN_LATENCY_RESULT_PARAMS params;
+            params.version = NV_VULKAN_LATENCY_RESULT_PARAMS_VER;
+            REQUIRE(NvAPI_Vulkan_GetLatency(vkDevice.get(), &params) != NVAPI_INCOMPATIBLE_STRUCT_VERSION);
+        }
+
+        SECTION("SetLatencyMarker returns OK") {
+            REQUIRE_CALL(*vkDevice, vkSetLatencyMarkerNV(_, _, _))
+                .WITH(_3->marker == VK_LATENCY_MARKER_PRESENT_START_NV)
+                .RETURN(VK_SUCCESS);
+
+            SetupResourceFactory(std::move(dxgiFactory), std::move(vk), std::move(nvml), std::move(lfx));
+
+            REQUIRE(NvAPI_Vulkan_InitLowLatencyDevice(vkDevice.get(), &signalSemaphoreHandle) == NVAPI_OK);
+
+            NV_VULKAN_LATENCY_MARKER_PARAMS params;
+            params.version = NV_VULKAN_LATENCY_MARKER_PARAMS_VER1;
+            params.markerType = VULKAN_PRESENT_START;
```

Comment:
I wonder if would be worth testing if unknown marker types are silently ignored…

Reply:
Yes, done as separate test.